app/helpers/mandatory_field_helper.rb

```diff
@@ -0,0 +1,90 @@
+    invalid_fields = []
+    mandatory_fields.each do |field|
+      if session[field].blank?
+        invalid_fields << { field: field.to_s,
+                            text: t("coronavirus_form.#{page}.#{field}.custom_error",
+                                    default: t("coronavirus_form.errors.missing_mandatory_text_field", field: t("coronavirus_form.#{page}.#{field}.label")).humanize) }
+      end
+    end
+    invalid_fields
+  end
+
+  def validate_date_fields(year, month, day, field)
+    return [] if year.blank? && month.blank? && day.blank?
+
+    invalid_fields = []
+    invalid_fields << { field: field.to_s, text: t("coronavirus_form.errors.missing_year", field: field).humanize } if year.blank?
+    invalid_fields << { field: field.to_s, text: t("coronavirus_form.errors.missing_month", field: field).humanize } if month.blank?
+    invalid_fields << { field: field.to_s, text: t("coronavirus_form.errors.missing_day", field: field).humanize } if day.blank?
+    unless(invalid_fields != [] || Date.valid_date?(year.to_i, month.to_i, day.to_i))
+      invalid_fields << { field: field.to_s, text: t("coronavirus_form.errors.invalid_date", field: field).humanize }
+    end
+    invalid_fields
+  end
+
+  def validate_radio_field(page, radio:, other: false)
+    if radio.blank?
+      return [{ field: page.to_s,
+                text: t(
+                  "coronavirus_form.#{page}.custom_select_error",
+                  default: t("coronavirus_form.errors.radio_field", field: t("coronavirus_form.#{page}.title")).humanize,
+                ) }]
+    end
+
+    if other != false && other.blank? && %w(Yes Other).include?(radio)
+      return [{ field: page.to_s,
+                text: t(
+                  "coronavirus_form.#{page}.custom_enter_error",
+                  default: t("coronavirus_form.errors.missing_mandatory_text_field", field: t("coronavirus_form.#{page}.title")).humanize,
+                ) }]
+    end
+
+    []
+  end
+
+  def validate_checkbox_field(page, values:, allowed_values:)
+    if values.blank? || values.empty?
+      return [{ field: page.to_s,
+                text: t(
+                  "coronavirus_form.#{page}.custom_select_error",
+                  default: t("coronavirus_form.errors.checkbox_field", field: t("coronavirus_form.#{page}.title")).humanize,
+                ) }]
+    end
+
+    if (values - allowed_values).any?
+      return [{ field: page.to_s,
+                text: t(
+                  "coronavirus_form.#{page}.custom_select_error",
+                  default: t("coronavirus_form.errors.missing_mandatory_text_field", field: t("coronavirus_form.#{page}.title")).humanize,
+                ) }]
+    end
+
+    []
+  end
+
+  def validate_date_order(start_date, end_date, field)
+    if end_date < start_date
+      [{ field: field.to_s, text: t("coronavirus_form.errors.date_order") }]
+    else
+      []
+    end
+  end
+
+  def validate_email_address(field, email_address)
```

Comment:
This file is called "mandatory field helper", but at least this validation isn't about mandatory fields. 

Maybe "field validation helper" would be better?

Reply:
Good point, I've change this.